enhancements/network/dpu-ovnkube-operator.md

```diff
@@ -0,0 +1,200 @@
+- [ ] Test plan is defined
+- [ ] Graduation criteria for dev preview, tech preview, GA
+- [ ] User-facing documentation is created in [openshift-docs](https://github.com/openshift/openshift-docs/)
+
+
+## Summary
+
+To facilitate the management of Nvidia BlueField-2 DPU, a two-cluster design is
+proposed. Under such design, a BlueFiled-2 card will be provisioned as a worker
+node of the ARM-based infra cluster, whereas the tenant cluster where the normal
+user applications run on, is composed of the X86 servers.
+
+The OVN-Kubernetes components are spread over the two clusters. On the tenant
+cluster side, the Cluster Network Operator is in charge of the management of the
+ovn-kube components. On the infra cluster side, we propose to create a new
+operator to be responsible for the life-cycle management of ovn-kube components
+and the necessary host network initialization on DPU cards.
+
+## Motivation
+
+Normally, the Cluster Network Operator watches the Network customer resource,
+then provision the ovn-kube components to the Openshift cluster.
+
+In the two-cluster design, some of the ovn-kube components have to run on BF2
+DPUs, but they are still part of the ovn-kube cluster running in the tenant
+cluster. To allow the ovn-kube components on different clusters to talk with
+each other, we need to ensure all the necessary configs and secrets are aligned
+with each other.
+
+### Goals
+
+- Design a new operator that
+  - manages the ovn-kube components that serve the tenant cluster.
+  - enable the switchdev mode on DPUs.
+  - add PF representor to the br-ex bridge, so that the x86 host can connect to
+  the cluster network of the tenant cluster.
+- Define the ovn-kube upgrade mechanism in such a deployment.
+
+### Non-Goals
+
+- Support for network plugins other than OVN-Kubernetes.
+- This enhancement does not address the two-cluster installation.
+- It is possible to have one infra cluster to serve more than one tenant cluster
+in the future. But it is not our initial goal.
+
+## Pre-requisites
+
+- This operator needs to access the apiserver of the tenant cluster, we need to
+  make sure that the pods network in the infra cluster can connect to the
+  apiserver of the tenant cluster in some way.
+- In the infra cluster, one custom MachineConfigPool shall be created for the
+  DPUs on which hardware offloading needs to be enabled.
+
+## Proposal
+
+We will create a namespaced CRD OVNKubeConfig.dpu.openshift.io for the DPU
+OVNKube Operator. One custom resource of this CRD shall be created for the
+tenant cluster in the infra cluster. The operator will take this custom resource
+as input to render and create the objects of the ovnkube DPU components. The
+operator will also create a MachineConfig CR which injects a systemd service to
+do necessary host configuration on DPUs.
+
+### User Stories
+
+### Risks and Mitigations
+
+## Design Details
+
+The DPU OVNKube Operator will fetch the necessary information from the
```

Comment:
Does this Operator need to handle kube API compatibility between infra and tenant clusters?

Reply:
Is that necessary? I assume both clusters shall in the same version.

Reply:
If we allow tenant and infra to be in different cluster version (which I think we should if not in the initial delivery), then yes.